.github/workflows/build-sdk.yaml

```diff
@@ -0,0 +1,41 @@
+name: Build and Cache SDK
+
+on:
+  pull_request:
+  merge_group:
+
+jobs:
+  build-cache-sdk:
+    name: Build and Cache SDK
+    runs-on: ubuntu-latest-4-cores
+    env:
+      NODE_OPTIONS: --max-old-space-size=14366
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: ".nvmrc"
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: yarn install --immutable
+
+      - name: Cache build artifacts
+        uses: actions/cache@v4
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-build-cache-deps-${{ hashFiles('yarn.lock') }}
```

Comment:
where is `build-cache-deps` coming from?
I don't know enough but i can see a pattern here the key is build-cache-xxx and the job is called `build-cache-sdk` is there another key/job exist with the word `deps` ?

Reply:
Not yet, that will be a follow up PR to use that cache once this workflow is merged to main.